components/edit/configure/imports/toolCatalogModal.tsx

```diff
@@ -7,16 +7,24 @@ import PropTypes from 'prop-types';
 interface ToolCatalogModalProps {
   tools: string[] | undefined;
   addTool: (tool: string) => void;
-  removeTool: (tool: string) => void;
 }
 
 const ToolCatalogModal: React.FC<ToolCatalogModalProps> = ({
   tools,
   addTool,
-  removeTool,
 }) => {
   const [isModalOpen, setIsModalOpen] = useState(false);
 
+  const addToolAndClose = useCallback(
+    (tool: string) => {
+      if (isModalOpen) {
```

Comment:
Nit: you can just do `setIsModalOpen(false);` It won't cause side-effect if you set to false even if `isModalOpen` is false already.